src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,326 @@
+        //if there is no child, then insert as child
+        if (!root->child) {
+            o->parent = root;
+            root->child = o;
+            return OS_SUCCESS;
+        }
+        root = root->child;
+
+        //precondition: root is the first in a non-empty list of children
+        //postcondition: root is an ancestor of o
+        while (strspn(o->name, root->name) < strlen(root->name)) {
+            if (!root->next) {
+                root->next = o;
+                o->prev = root;
+                o->parent = root->parent;
+                return OS_SUCCESS;
+            }
+            root = root->next;
+        }
+
+    }
+    assert(0);
+    return OS_FS_ERROR;
+}
+
+int32 open(const char *path) {
+    assert(fs[0].root && !fs[1].root);
+    struct fsobj *o = fs[0].root;
+    o = find(path, o);
+    if (!o) return OS_FS_ERROR;
+    o->refcnt++;
+    return o->filedes;
+}
+
+int32 close(int32 filedes) {
+    if (files[filedes + 1].filedes != filedes)   return OS_FS_ERROR;
+    if (files[filedes + 1].refcnt < 1)           return OS_FS_ERROR;
+    files[filedes + 1].refcnt--;
+    return OS_FS_SUCCESS;
+}
+
+int32 read(int32  filedes, void *buffer, uint32 nbytes) {
+    if (!buffer)                                 return OS_FS_ERR_INVALID_POINTER;
+    if (files[filedes + 1].filedes != filedes)   return OS_FS_ERROR;
+    if (files[filedes + 1].refcnt < 1)           return OS_FS_ERROR;
+    if (!nbytes)                                 return 0;
+    memcpy(buffer, files[filedes + 1].data, nbytes);
+    return nbytes;
+}
+
+struct fsobj *find(const char *path, struct fsobj *root)
+{
+    assert(root);
+    assert(strlen(root->name) < strlen(path));
+    while (1) {
+        assert(strcmp(root->name, path) !=0 );
+        assert(strspn(path, root->name) >= strlen(root->name));
+        if (!root->child) return NULL;
+
+        root = root->child;
+
+        while (strspn(path, root->name) < strlen(root->name)) {
+            if (!strcmp(root->name, path)) return root;
+            if (!root->next) return NULL;
+            root = root->next;
+        }
+
+    }
+    assert(0);
+    return NULL;
+}
+
+/******************************************************************************
+** fs Level Methods
+******************************************************************************/
+
+uint32 mount(const char *devname, char *mountpoint)
+{
+    uint32 i = 0;
+    while (i < MAX_NUM_FS && fs[i].devname != NULL) {
+        if (!strcmp(fs[i].devname, devname)) {
+            fs[i].mountpoint = mountpoint;
+            return OS_FS_SUCCESS;
+        }
+        i++;
+    }
+    OS_printf("mount failed\n");
+    return OS_FS_ERROR;
+}
+
+uint32 unmount(const char *mountpoint) {
+    uint32 i = 0;
+    while (i < MAX_NUM_FS) {
+        if (mountpoint == NULL);
+        else if (!strcmp(fs[i].mountpoint, mountpoint)) {
+            fs[i].mountpoint = NULL;
+            return OS_FS_SUCCESS;
+        }
```

Comment:
Collapse this into one if statement. You should never have an empty if statement.